dagshub/data_engine/model/datasource.py

```diff
@@ -334,6 +334,38 @@ def as_of(self, time: Union[float, datetime.datetime]) -> "Datasource":
         new_ds._query.as_of = to_timestamp(time)
         return new_ds
 
+    def order_by(self, *args: Union[str, Tuple[str, bool], Tuple[str, str]]):
+        """
+        Sort the query result by the specified fields.
+
+        Args:
+            Fields to sort by. Can be either of:
+                - Name of the field to sort by: ``"field"``.
+                - A tuple of ``(field_name, ascending)``: ``("field", True)``.
+                - A tuple of ``(field_name, "asc"|"desc")``: ``("field", "asc")``.
+
+        Examples::
+
+            ds.sort_by("size", ("date", "desc")).all()
+            ds.sort_by("size", ("date", False)).all()
+        """
+        new_ds = self.__deepcopy__()
+        new_ds.get_query().order_by = []
+        for arg in args:
+            if isinstance(arg, str):
+                new_ds.get_query().order_by.append({"field": arg, "order": "ASC"})
+            else:
+                if len(arg) != 2:
+                    raise RuntimeError(f"Invalid sort argument {arg}")
+                if isinstance(arg[1], bool):
+                    order = "ASC" if arg[1] else "DESC"
+                elif isinstance(arg[1], str) and arg[1].upper() in ["ASC", "DESC"]:
+                    order = arg[1].upper()
+                else:
+                    raise RuntimeError(f"Invalid sort argument {arg}")
+                new_ds.get_query().order_by.append({"field": arg[0], "order": order})
```

Comment:
Extending as you go might lead to broken ordering later (even though I see you clearing it here right now).
Maybe have an orders list and then set `new_ds.get_query().order_by = orders` in the end

Reply:
I guess since raising an error will not return `new_ds` it actually doesn't matter. nevermind

Reply:
I actually like your suggestion.